nucypher/blockchain/eth/signers.py

```diff
@@ -191,3 +216,93 @@ def unlock_account(self, account: str, password: str, duration: int = None) -> b
     @validate_checksum_address
     def lock_account(self, account: str):
         return True
+
+
+class KeyStoreSigner(Signer):
+
+    IS_SIGNER_LOCAL = True
+
+    def __init__(self, uri: str):
+        super().__init__()
+
+        decoded_uri = urlparse(uri)
+        if decoded_uri.scheme not in ('key', 'keystore') or decoded_uri.netloc:
+            raise cls.InvalidSignerURI(uri)
+
+        with open(decoded_uri.path) as file_obj:
+            key = json.load(file_obj)
+
+        address = key['address']
```

Comment:
Is there any situation in which `KeyError` might be raised here?

Reply:
Yes but only if the key store file is corrupt or invalid.

Reply:
Should we be more explicit with that error instead of a `KeyError`? We could do a `get()` and if the returned address is `None`, throw a `ValueError` indicating file may be invalid or corrupt.

Reply:
Constructing a KeyStoreSigner can raise the following non-exclusive exceptions:

- `FileNotFoundError` or `PermissionError` in `open`
- `JSONDecodeError' in `json.load`
- `TypeError` in `key['address']` if the deserialised JSON was not an object (i.e. array, string, number, etc.)
- `KeyError` in `key['address']` if there's no address key in the object
- `ValueError` raised explicitly if `key['address']` isn't a valid Ethereum address

There will be other possible exceptions that could conceivably crop up here.
The question is whether there is anything intelligent or useful `KeyStoreSigner.__init__()` can do about it here or whether it is better placed to interpret the exception and add useful information for the caller.

It seems to me the answer is no here but I of course could be mistaken - the fact that any exception occurs here tells the caller construction of the signer object failed which is the key information. Unless there is a generalised exception hierarchy specifically intended to allow callers to effectively handle correctable errors programatically in a general way then I see no point in adding additional code to normalise python exceptions.

But that is just my humble opinion!

Reply:
The other exceptions are clearer to me. If we are trying to be clear about the source of this problem, in my mind, a `KeyError` is much less clear than a `ValueError` with a message that the file is invalid or corrupt. 

That being said, this is not a make or break for the PR - just a thought.